### ISSUE TITLE
Events don't occur after stage 1 + 'Give Item' events give 0 of item

### ISSUE THREAD

**Issue by unamina** [bryantBaumgartner/chaosMode#2]
Enjoying the mod so far!

Unfortunately it seems like events just cease to function after you leave Stage 1.

As well, Item events look like they grant 0 of the item. As shown in the picture, I'm given the item, but I'm given 0 of the item.

![Risk_of_Rain_2_4xkyku8slf](https://user-images.githubusercontent.com/101493979/158049045-30646949-26aa-4179-a291-55678da922bc.png)

**Comment by bryantBaumgartner** [comment]
Interesting.  I have an idea as to what's causing both of those issues and I'll try to take a look and push a fix later today.
Did you change any of the Config settings or are you using the mod with default settings?

**Comment by bryantBaumgartner** [comment]
I just uploaded a patch that should prevent items from being distributed as 0, however I'm still looking into why the Events are stopping after Stage 1.

**Comment by unamina** [comment]
> Interesting. I have an idea as to what's causing both of those issues and I'll try to take a look and push a fix later today. Did you change any of the Config settings or are you using the mod with default settings?

Default settings

**Comment by unamina** [comment]
I ran a couple tests just sitting on stage 1 for a while, and it doesn't seem to stop specifically because of stage changes, it seems random. I had one run where events stopped after 6 minutes, but another where they didn't stop until 14 minutes, all on default settings. 

Hope that helps pinpoint the issue

**Comment by bryantBaumgartner** [comment]
I've found a potential cause of the bug, and a couple ways that I'm going to try to patch it.

One more question, before the Events stopped, was the game trying to spawn/did the game successfully spawn any Elite enemies?

**Comment by bryantBaumgartner** [comment]
I took a look and found that whenever the game tries to give the player an Equipment, the mod freezes.

I removed Equipment from the random distribution pool so now they only drop from Chests as normal. Testing seemed to show the mod working correctly now. I'll keep this thread open for a bit in case you notice any more issues like this.

**Comment by unamina** [comment]
It looks like it still tries to give you equipment for whatever reason, but at least it doesn't freeze up the mod, and it doesn't really seem to cause any issues, it just fails to give the item obviously.

![Risk_of_Rain_2_DRAROwcCQO](https://user-images.githubusercontent.com/101493979/158151234-6000ddbc-b7f0-4fcb-a593-66b29d5b0157.jpg)
!

> I've found a potential cause of the bug, and a couple ways that I'm going to try to patch it.
> 
> One more question, before the Events stopped, was the game trying to spawn/did the game successfully spawn any Elite enemies?

It was spawning elites with Events pre-update, and it still is post-update, in case you were still needing that info

**Comment by bryantBaumgartner** [comment]
I was able to prevent the mod from crashing due to this bug a few patches ago, but after some updates I realized that I had mismatched which items the game gives you. Basically, when the game tried to give you a Lunar Item for example, it pulled from the pool of Equipment. This was why patching Equipment drops didn't seem to actually fix the problem.

I reworked the drop table system so this problem should be gone for good, and I may be able to add Equipment back to the pool of randomly given items soon.
